ludwig/datasets/dataset_config.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) 2022 Predibase, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from dataclasses import dataclass, field
+from typing import Dict, List, Optional, Union
+
+
+@dataclass
+class DatasetConfig:
+    """The configuration of a Ludwig dataset."""
+
+    # The version of the dataset.
+    version: str
+
+    # The name of the dataset. Make this a valid python module name, should not contain spaces or dashes.
+    name: str
+
+    # The readable description of the dataset
+    description: str = ""
+
+    # The kaggle competition this dataset belongs to, or None if this dataset is not hosted by a Kaggle competition.
+    kaggle_competition: Optional[str] = None
+
+    # The kaggle dataset ID, or None if this dataset if not hosted by Kaggle.
+    kaggle_dataset_id: Optional[str] = None
+
+    # The list of URLs to download.
+    download_urls: Union[str, List[str]] = field(default_factory=list)
+
+    # The list of file archives which will be downloaded. If download_urls contains a filename with extension, for
+    # example https://domain.com/archive.zip, then archive_filenames does not need to be specified.
+    archive_filenames: Union[str, List[str]] = field(default_factory=list)
+
+    # The names of files in the dataset (after extraction). Glob-style patterns are supported, see
+    # https://docs.python.org/3/library/glob.html
+    dataset_filenames: Union[str, List[str]] = field(default_factory=list)
+
+    # If the dataset contains separate files for training, testing, or validation. Glob-style patterns are supported,
+    # see https://docs.python.org/3/library/glob.html
+    train_filenames: Union[str, List[str]] = field(default_factory=list)
+    validation_filenames: Union[str, List[str]] = field(default_factory=list)
+    test_filenames: Union[str, List[str]] = field(default_factory=list)
+
+    # If the dataset contains additional referenced files or directories (ex. images or audio) list them here and they
+    # will be copied to the same location as the processed dataset. Glob-style patterns are supported,
+    # see https://docs.python.org/3/library/glob.html
+    preserve_paths: Union[str, List[str]] = field(default_factory=list)
+
+    # Optionally verify integrity of the dataset by providing sha256 checksums for important files. Maps filename to
```

Comment:
What happens when the sha256 is different? Is a warning logged, or does the dataset simply fail to load?

Reply:
It will fail with ValueError("Checksum mismatch for file....

Added more detail in the description.